File: board-admin-backend/src/main/java/com/portSrilanka/board_admin_backend/dto/user/UserRequest.java
package com.portSrilanka.board_admin_backend.dto.user;

import com.portSrilanka.board_admin_backend.enums.BoardType;
import com.portSrilanka.board_admin_backend.enums.AccessProfile;
import lombok.Data;

@Data
public class UserRequest {
    private String salutation;
    private String firstName;
    private String lastName;
    private String displayName;
    private String boardEmail;
    private String officeEmail;
    private String officeNumber;
    private String mobileNumber;
    private String jobTitle;
    private String profilePictureUrl;
    private boolean twoStepEnabled;
    private String role;
    private BoardType boardType;
    private AccessProfile accessProfile;
}

File: board-admin-frontend/frontend/lib/features/users/model/user_request.dart
class UserRequest {
  final String firstName;
  final String lastName;
  final String? displayName;
  final String email;
  final String? officeEmail;
  final String? officeNumber;
  final String? mobileNumber;
  final String? jobTitle;
  final String role;
  final String boardType;
  final String accessProfile;
  final bool twoStepEnabled;

  UserRequest({
    required this.firstName,
    required this.lastName,
    this.displayName,
    required this.email,
    this.officeEmail,
    this.officeNumber,
    this.mobileNumber,
    this.jobTitle,
    required this.role,
    required this.boardType,
    required this.accessProfile,
    required this.twoStepEnabled,
  });

  Map<String, dynamic> toJson() {
    return {
      'firstName': firstName,
      'lastName': lastName,
      'displayName': displayName,
      'boardEmail': email,
      'officeEmail': officeEmail,
      'officeNumber': officeNumber,
      'mobileNumber': mobileNumber,
      'jobTitle': jobTitle,
      'role': role,
      'twoStepEnabled': twoStepEnabled,
      'boardType': boardType,
      'accessProfile': accessProfile,
    };
  }
}
